src/Hosting/Hosting/src/Internal/HttpMetricsTagsFeature.cs

```diff
@@ -10,4 +10,19 @@ internal sealed class HttpMetricsTagsFeature : IHttpMetricsTagsFeature
     ICollection<KeyValuePair<string, object?>> IHttpMetricsTagsFeature.Tags => TagsList;
 
     public List<KeyValuePair<string, object?>> TagsList { get; } = new List<KeyValuePair<string, object?>>();
+
+    // Cache request values when request starts. These are used when writing metrics when the request ends.
+    // This ensures that the tags match between the start and end of the request.
+    public string? Method { get; set; }
+    public string? Scheme { get; set; }
+    public string? Protocol { get; set; }
+
+    public void Reset()
```

Comment:
Unused?

Reply:
Thanks, I meant for this to be used to clear out cached values.